### PR TITLE
Update 1.9.0-rc1 changelog

### PR DESCRIPTION
**Pull request by mdeggies**
Update the changelog to reflect improvements introduced with the 1.9.0-rc1 release

### CODE REVIEW

**Comment by mdeggies** [comment]
Adding @kpenfound and @mladlow as reviewers- feel free to update if needed. The changelog links should work once the CI changes are added to `main`, but this probably shouldn't be merged until then.

**Comment by mladlow** [comment]
@mdeggies if these changes were manually edited, they'll be overwritten by the changelog generation for 1.9.0 - could you add a corresponding changelog file?

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ IMPROVEMENTS:
 * core: Upgrade github.com/gogo/protobuf [[GH-12255](https://github.com/hashicorp/vault/pull/12255)]
 * core: build with Go 1.17, and mitigate a breaking change they made that could impact how approle and ssh interpret IPs/CIDRs [[GH-12868](https://github.com/hashicorp/vault/pull/12868)]
 * core: observe the client counts broken down by namespace for partial month client count [[GH-12393](https://github.com/hashicorp/vault/pull/12393)]
+* core: Artifact builds will now only run on merges to the release branches or to `main`
+* core: The [dockerfile](https://github.com/hashicorp/vault/blob/main/Dockerfile) that is used to build the vault docker image available at [hashicorp/vault](https://hub.docker.com/repository/docker/hashicorp/vault) now lives in the root of this repo, and the entrypoint is available under [.release/docker/docker-entrypoint.sh](https://github.com/hashicorp/vault/blob/main/.release/docker/docker-entrypoint.sh)(
```
Comment (mladlow) [inline CHANGELOG.md:66]:
```suggestion
* core: The [dockerfile](https://github.com/hashicorp/vault/blob/main/Dockerfile) that is used to build the vault docker image available at [hashicorp/vault](https://hub.docker.com/repository/docker/hashicorp/vault) now lives in the root of this repo, and the entrypoint is available under [.release/docker/docker-entrypoint.sh](https://github.com/hashicorp/vault/blob/main/.release/docker/docker-entrypoint.sh)
```

**Review by mladlow — APPROVED** [review]

**Comment by mladlow** [comment]
In the changelog file I changed the prefix to be "core/build" so they'd all get grouped together.

File: CHANGELOG.md [hunk 2]
```diff
@@ -62,6 +62,10 @@ IMPROVEMENTS:
 * core: Upgrade github.com/gogo/protobuf [[GH-12255](https://github.com/hashicorp/vault/pull/12255)]
 * core: build with Go 1.17, and mitigate a breaking change they made that could impact how approle and ssh interpret IPs/CIDRs [[GH-12868](https://github.com/hashicorp/vault/pull/12868)]
 * core: observe the client counts broken down by namespace for partial month client count [[GH-12393](https://github.com/hashicorp/vault/pull/12393)]
+* core: Artifact builds will now only run on merges to the release branches or to `main`
+* core: The [dockerfile](https://github.com/hashicorp/vault/blob/main/Dockerfile) that is used to build the vault docker image available at [hashicorp/vault](https://hub.docker.com/repository/docker/hashicorp/vault) now lives in the root of this repo, and the entrypoint is available under [.release/docker/docker-entrypoint.sh](https://github.com/hashicorp/vault/blob/main/.release/docker/docker-entrypoint.sh)
+* core: The vault linx packaging service configs and pre/post install scripts are now available under [.release/linux](https://github.com/hashicorp/vault/blob/main/.release/linux)
```
Comment (kpenfound) [inline CHANGELOG.md:67]:
```suggestion
* core: The vault linux packaging service configs and pre/post install scripts are now available under [.release/linux](https://github.com/hashicorp/vault/blob/main/.release/linux)
```

**Review by kpenfound — APPROVED** [review]
Looks good!

File: CHANGELOG.md (hunk 2 above)
Comment (mdeggies) [reply CHANGELOG.md:67]: Haha thank you!

**Comment by mladlow** [comment]
Would you mind waiting to merge this until we do the GA release this week, and rebasing off the stuff that's going to be pushed to main instead?

**Pull request merged**